packages/@aws-cdk-containers/ecs-service-extensions/lib/extensions/publisher.ts

```diff
@@ -0,0 +1,169 @@
+import * as ecs from '@aws-cdk/aws-ecs';
+import * as iam from '@aws-cdk/aws-iam';
+import * as sns from '@aws-cdk/aws-sns';
+import { Service } from '../service';
+import { Container } from './container';
+import { ContainerMutatingHook, ServiceExtension } from './extension-interfaces';
+
+// Keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from '@aws-cdk/core';
+
+/**
+ * An interface that will be implemented by all the resources that can be published events to.
+ */
+export interface IPublisher {
+  publish(taskDefinition: ecs.TaskDefinition): void;
+
+  envVarKey(): string;
+
+  envVarValue(): string;
+}
+
+/**
+ * The settings for the `PublisherTopic` class.
+ */
+export interface PublisherTopicProps {
+  /**
+   * The SNS Topic to publish events to.
+   */
+  readonly topic: sns.ITopic;
+
+  /**
+   * The accounts allowed to subscribe to the given `topic`.
+   *
+   * @default none
+   */
+  readonly allowedAccounts?: string[];
+}
+
+/**
+ * The `PublisherTopic` class represents SNS Topic resource that can be published to by the parent service.
+ */
+
+export class PublisherTopic implements IPublisher {
+  public readonly topic: sns.ITopic;
+
+  constructor(props: PublisherTopicProps) {
+    this.topic = props.topic;
+
+    if (props.allowedAccounts) {
+      const principals = [];
+      for (const account of props.allowedAccounts) {
+        principals.push(new iam.AccountPrincipal(account));
+      }
+
+      this.topic.addToResourcePolicy(new iam.PolicyStatement({
+        actions: ['sns:Subscribe'],
+        resources: [this.topic.topicArn],
+        principals,
+      }));
+    }
+  }
+
+  public publish(taskDefinition: ecs.TaskDefinition) {
+    this.topic.grantPublish(taskDefinition.taskRole);
+  }
+
+  public envVarKey(): string {
+    return this.topic.node.id;
+  }
+
+  public envVarValue(): string {
+    return this.topic.topicArn;
+  }
+}
```

Comment:
hmm these function only contain one line logic. I wonder if we want this extension since it doesn't seem to bring a lot of convenience for users. Could you remind me how users are going to publish to a topic without this extension, so that we can better evaluate the value of bring this extra public extension 🙏

Reply:
This extension helps set up the publish permissions for SNS Topics and optionally restrict subscribe permissions to particular accounts. The simplest use case for setting up the publish permissions can be done by calling `topic.grantPublish(taskDefinition.taskRole)` . 

This extension might seem to be more useful after we add more `IPublisher` resources perhaps? (like publishing to an EventBus)